airflow/models.py

```diff
@@ -346,10 +346,10 @@ def kill_zombies(self, session=None):
 
                     # now set non db backed vars on ti
                     ti.task = task
-                    ti.test_mode = False
+                    ti.test_mode = configuration.getboolean('core', 'unit_test_mode')
```

Comment:
@bolkedebruin I believe that if `test_mode` is true then that `ti` would never have been picked by `kill_zombie` code, so I'd it `False` initially.